chemprop/uncertainty/calibrator.py

```diff
@@ -68,13 +70,59 @@ def apply_calibration(self, preds, uncs) -> Tensor:
 
 @UncertaintyCalibratorRegistry.register("platt")
 class PlattCalibrator(UncertaintyCalibrator):
-    def calibrate(self, preds, uncs, targets, mask) -> Tensor:
-        ...
-        return
-
-    def apply_calibration(self, preds, uncs) -> Tensor:
-        ...
-        return
+    """
+    A calibration method for classification datasets based on the Platt scaling algorithm.
+    As discussed in https://arxiv.org/abs/1706.04599 and Platt, J. (2000). Probabilistic outputs for
+    support vector machines and comparison to regularized likelihood methods. In A. Smola, P. 
+    Bartlett, B. Schölkopf, & D. Schuurmans (Eds.), Advances in large margin classifiers. Cambridge:
+    MIT Press.
+    In Platt's paper, he suggests using the number of positive and negative examples in the dataset
+    used to train the model to adjust the value of target probabilities used to fit the parameters. 
+    """
+
+    def calibrate(self, preds, targets, mask, training_targets: None | Tensor = None) -> Tensor:
+        if not torch.logical_or((targets[mask] == 0), (targets[mask] == 1)).all():
+            raise ValueError("Platt scaling is only implemented for binary classification tasks.")
```

Comment:
```suggestion
        if (targets[mask] != 0).any() or (targets[mask] == 1).any():
            raise ValueError("Platt scaling is only implemented for binary classification tasks! Input tensor must contain only 0's and 1's.")
```

Reply:
a general point, `any()` is typically faster than `all()` because it returns as soon as a `True` value is encountered. Here, you just want to check for the presence _any_ non-{0,1} entries, which is equivalent to checking that _all_ entries are in {0, 1}.